src/proj.rs

```diff
@@ -498,6 +500,56 @@ pub struct Proj {
 }
 
 impl Proj {
+    /// Create a coordinate metadata object to be used in coordinate operations.
+    ///
+    /// This creates a coordinate metadata object that can be used in coordinate operations,
+    /// such as transformations. The coordinate metadata object contains information
+    /// about the coordinates, such as the epoch they are referenced to.
+    ///
+    /// # Note
+    /// Only **transformation objects** (e.g. those created by calling [`new()`](fn@Proj::new())) can be
```

Comment:
If you do the wrong thing and try to create coordinate metadata for a pipeline, than [you'll hit this](https://github.com/georust/proj/discussions/220#discussioncomment-12833479), right?: 

> called `Result::unwrap()` on an `Err` value: ProjError("Unknown error (code 4096)")


Should we try to be a "better proj than proj" and encode this in the type system?

I'm ok either way for now.

Reply:
I played around with that idea a bit last night. The fundamental issue is that _we_ don't distinguish between a pipeline and a standard transformation object (neither does libproj), which means we can't get fancy with a bound (I don't think…): everything's a Proj. We have some runtime options:
1. Catch the error and return something more informative;
2. Check whether the input is a pipeline using proj_info, and return an informative error if kind is Some("Pipeline").

Reply:
Having a runtime check like you described could be a good compromise. We could at least produce a useful error message.

Reply:
I am very much open to something compile-time. But it would be very much a breaking change, right?

Reply:
Oh hell yes. We could maybe minimize it by having a default? But definitely breaking.

Reply:
I'm also fine to not do it (or do it later).

Reply:
OK, we've got a more informative error message. I think rather than breaking everyone for the sake of an operation that isn't required by most users, this will do for now.